plugins/polio/api/campaigns/campaigns.py

```diff
@@ -579,6 +580,26 @@ class CalendarCampaignSerializer(CampaignSerializer):
     """This serializer contains juste enough data for the Calendar view in the web ui. Read only.
     Used by both anonymous and non-anonymous user"""
 
+    class NestedSubactivitySerializer(serializers.ModelSerializer):
+        class NestedScopeSerializer(serializers.ModelSerializer):
+            class NestedGroupSerializer(GroupSerializer):
+                class Meta:
+                    model = Group
+                    fields = ["id"]
+
+            class Meta:
+                model = SubActivityScope
+                fields = ["group", "vaccine"]
+
+            group = NestedGroupSerializer()
+
+        scopes = NestedScopeSerializer(many=True)
+        round_number = serializers.IntegerField(source="round.number")
+
+        class Meta:
+            model = SubActivity
+            fields = ["name", "start_date", "end_date", "scopes", "id", "vaccine_names", "round_number"]
```

Comment:
Nitpick : I know it's from before but this is not very readable. I don't personally like to put class inside class inside class ...

Reply:
TBH me neither